Deploy downcast<> in ElementData

AlexsanderDamaceno opened the pull request:
#### aee51a1d1c98e00026ddcd9637f251c710bc65af
<pre>
Deploy more downcast&lt;&gt; in Element
<a href="https://bugs.webkit.org/show_bug.cgi?id=281894">https://bugs.webkit.org/show_bug.cgi?id=281894</a>

Reviewed by NOBODY (OOPS!).

Deploy more downcast&lt;&gt; in Element

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::synchronizeAllAttributes const):
(WebCore::Element::synchronizeAttribute const):
</pre>

cdumez commented on Source/WebCore/dom/ElementData.cpp:
```diff
@@ -154,8 +154,8 @@ UniqueElementData::UniqueElementData(const ShareableElementData& other)
 Ref<UniqueElementData> ElementData::makeUniqueCopy() const
 {
     if (isUnique())
```
This would be better written (could be slightly more efficient) as:
```
if (auto* uniqueData = dynamicDowncast<const UniqueElementData>(*this))
    return return adoptRef(*new UniqueElementData(*uniqueData));
return adoptRef(*new UniqueElementData(downcast<const ShareableElementData>(*this)));
```

Otherwise, we're doing the `isUnique()` check twice.

AlexsanderDamaceno replied:
@cdumez well noted, thank you for point out this, changed.

cdumez reviewed (approved):
lgtm

AlexsanderDamaceno committed:
Deploy downcast<> in ElementData
https://bugs.webkit.org/show_bug.cgi?id=281475

Reviewed by Chris Dumez.

Deploy downcast<> in ElementData

* Source/WebCore/dom/ElementData.cpp:
(WebCore::ElementData::makeUniqueCopy const):

Canonical link: https://commits.webkit.org/285259@main
